Added note about custom domain in fetched URL for Session Replay integrations (#17922)
DataDog/documentation: content/en/real_user_monitoring/guide/connect-session-replay-to-your-third-party-tools.md

To fetch the URL for the current user session's recording, use the following snippet, depending on the installation method you used to set up RUM:

> **Note**: Although providing a value for `subdomain` when fetching the user session's recording URL is optional, you should provide it if you're accessing Datadog through a custom subdomain and want to see the custom domain in the URL that gets returned.

Reviewer: ```suggestion
**Note**: Providing a value for `subdomain` when fetching the user session's recording URL is optional, but must be provided if you're accessing Datadog through a custom subdomain and want to see the custom domain in the URL that gets returned.
```

Reviewer: [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119) is the source of truth for determining the meanings of words like `should` and `must` -- I'm guessing in this case that the correct word is `must`, but feel free to ignore this suggestion if that's incorrect.

Author: Thank you @buraizu for the suggestion. It is now much clearer!
